stabbity.lic

```diff
@@ -0,0 +1,253 @@
+        fput('advance')
+        pause 3
+      when 'You\'ll need to stand up first'
+        fix_standing
+        hide?
+      when 'flying out of reach', 'to stop flying before'
+        echo "*** Switch to thrown -- add this MOB to your use_thrown_on list ***" if @debug
+        use_weapon 'thrown'
+        kill_thrown
+        return
+      when 'Roundtime', /\[You\'re /
+        return if npc_dead?
+        waitrt?
+        dodge_arena_trap if arena_mode?
+        hide?
+      end
+    end
+  end
+
+  def kill_thrown
+    action = @settings['stabbity']['thrown_action'].nil? ? 'throw' : @settings['stabbity']['thrown_action']
+    recover_command = @settings['stabbity']['thrown_invoke'] ? 'invoke' : "get #{@thrown_weapon}"
+    fix_standing
+    while @npc_is_alive
+      case bput(action, 'Roundtime', 'What are you trying', 'There is nothing else')
+      when 'What are you trying'
+        bput(recover_command, 'You pick up', 'suddenly leaps')
+        waitrt?
+      when 'Roundtime'
+        waitrt?
+        bput(recover_command, 'You pick up', 'suddenly leaps')
+        return if npc_dead?
+        waitrt?
+      when 'There is nothing else'
+        @npc_is_alive = false
+        return
+      end
+    end
+  end
+
+  def get_npcs
+    DRRoom.npcs - @ignore_npcs
+  end
+
+  def npc_dead?
+    if DRRoom.dead_npcs.include?(@target) || get_npcs.empty?
+      echo "*** Target #{@target} has died ***" if @debug
+      @npc_is_alive = false
+      return true
+    else
+      return false
+    end
+  end
+
+  def using_right_weapon?
+    if @alternate_weapon_npcs.include? @target
+      echo "*** Switching to alternate weapon ***" if @debug
+      use_weapon 'alternate'
+      hide?
+    elsif @thrown_weapon_npcs.include? @target
+      echo "*** Switching to thrown weapon ***" if @debug
+      use_weapon 'thrown'
+    elsif @current_weapon != 'preferred'
+      echo "*** Switching to preferred weapon" if @debug
+      use_weapon
+      hide?
+    end
+  end
+
+  def use_weapon(type = 'preferred')
+    echo "*** Already on right type ***" if @debug && @current_weapon == type
+    return if @current_weapon == type
+    case type
+    when 'preferred'
+      @current_weapon = 'preferred'
+      return if right_hand =~ /#{@preferred_weapon}/
+      @equipmanager.stow_weapon
+      @equipmanager.wield_weapon(@preferred_weapon) unless @preferred_weapon.empty?
+    when 'alternate'
+      @current_weapon = 'alternate'
+      return if right_hand =~ /#{@alternate_weapon}/
+      @equipmanager.stow_weapon
+      @equipmanager.wield_weapon(@alternate_weapon) unless @alternate_weapon.empty?
+    when 'thrown'
+      @current_weapon = 'thrown'
+      return if right_hand =~ /#{@thrown_weapon}/
+      @equipmanager.stow_weapon
+      @equipmanager.wield_weapon(@thrown_weapon)
+    end
+    echo "*** Changed weapon type to #{type} ***" if @debug
+  end
+
+  def watch_target
+    case bput("watch #{@target}", 'you could try to pedal', 
```

Comment:
You could use a regex to simplify this

Reply:
Stole that code from someone else. Looking at it, I believe it could be simplified to one line, but would appreciate your thoughts on this approach:
```
  def watch_target
    # Actions as of DuskRuin 2019: pedal, bob, duck, jump, lean, cower
    @arena_trap = bput("watch #{@target}", /(?:pedal|bob|duck|jump|lean|cower)/)
  end
```

Reply:
@rpherbig ^

Reply:
Looks good to me